### PR TITLE
CORENET-6742: OTE: enable and implement [Feature:Service] suite

### PR DESCRIPTION
**Pull request by jluhrsen**
Removes [Feature:Service] from the Disabled:Unimplemented list in test annotation rules.

Regenerates test annotations, enabling 37 Service-related E2E tests including NodePort,
ClusterIP, LoadBalancer, session affinity, and hairpin SNAT test coverage.

## Summary by CodeRabbit

* **Tests**
  * Expanded the “informing” set with additional OpenShift Service conformance test identifiers (including MetalLB and NodePort behaviors).
  * Added/strengthened early skipping for Service e2e tests when external container infrastructure is unavailable.
  * Improved log capture during IP-fragment testing and refined MetalLB speaker pod discovery for suite setup.
* **Bug Fixes**
  * Replaced “not implemented” panics/nil assumptions with explicit, user-facing errors when external-infra dependencies are uninitialized.
* **Chores**
  * Updated annotation rule mappings by removing the Service entry under the disabled/unimplemented key.

### CODE REVIEW

File: test/e2e/e2e.go [hunk 1]
```diff
@@ -428,9 +429,53 @@ func getPod(f *framework.Framework, podName string) *v1.Pod {
...
+	_, err := infraprovider.Get().PrimaryNetwork()
+	if err != nil {
+		// Only skip if this is the known "infrastructure not configured" error.
+		// Other errors indicate a provider regression and should fail the test.
+		if strings.Contains(err.Error(), "external container infrastructure not available") {
+			e2eskipper.Skipf("Skipping test - requires external container infrastructure: %v", err)
+		}
+		// If it's a different error, let it propagate as a test failure
+		framework.ExpectNoError(err, "PrimaryNetwork() failed with unexpected error")
+	}
```
[the remark this answers is not in the document]
Comment (jluhrsen) [reply test/e2e/e2e.go:498]: Thank you for the suggestion. We've decided not to implement a typed sentinel error for the following reasons:

  1. Module boundaries: The test/e2e and openshift/test packages are in separate Go modules with replace directives.
  Exposing a sentinel error would require either cross-module dependencies or duplicating the error definition,
  adding unnecessary complexity.
  2. Current implementation is safe: The existing string check achieves the same safety goal:
    - ✅ Only skips on the specific expected error message ("external container infrastructure not available")
    - ✅ Fails tests on unexpected errors (prevents masking regressions)
    - ✅ The error message is a constant in the provider, so it's stable
  3. String matching is explicit: strings.Contains(err.Error(), "external container infrastructure not available")
  makes it clear what we're checking for, whereas errors.Is() would require developers to track down the sentinel
  definition across module boundaries.

  The current approach provides type safety through specificity (exact message matching) rather than Go's type
  system, which is appropriate for this cross-module boundary case.

File: test/e2e/service.go [hunk 2]
```diff
@@ -978,6 +978,13 @@ var _ = ginkgo.Describe("Services", feature.Service, func() {
 		})
 
 		ginkgo.It("should listen on each host addresses", func() {
+			// Skip if external container infrastructure is not available
+			// This test requires bare-metal cluster infrastructure with external containers
+			_, err := infraprovider.Get().PrimaryNetwork()
+			if err != nil {
+				e2eskipper.Skipf("Skipping test - requires external container infrastructure: %v", err)
+			}
```
[the remark this answers is not in the document]
Comment (jluhrsen) [reply test/e2e/service.go]: does it look ok now?
